docker/start_server.sh

```diff
@@ -0,0 +1,118 @@
+#!/bin/bash
+
+function set_defaults() {
+    DATABASE_PORT=${DATABASE_PORT:-3306}
+    DATABASE=${DATABASE:-darkflame}
+
+    if [[ -z $DATABASE_PASSWORD ]]; then
+        USE_DATABASE_PASSWORD="No"
+    else
+        USE_DATABASE_PASSWORD="Yes"
+    fi
+
+    CHAT_SERVER_PORT=${CHAT_SERVER_PORT:-2005}
+    MAX_CLIENTS=${MAX_CLIENTS:-999}
+
+    echo "Start server with configuration:"
+    echo "===== Database Config ========="
+    echo "Database: $DATABASE"
+    echo "Database host: $DATABASE_HOST"
+    echo "Database port: $DATABASE_PORT"
+    echo "Database user: $DATABASE_USER"
+    echo "Database password set: $USE_DATABASE_PASSWORD"
+    echo "===== Other settings =========="
+    echo "Chat server port: $CHAT_SERVER_PORT"
+    echo "Max clients: $MAX_CLIENTS"
+}
+
+function check_sql_connection() {
+    until echo '\q' | mysql -h"$DATABASE_HOST" -P"$DATABASE_PORT" -u"$DATABASE_USER" -p"$DATABASE_PASSWORD" $DATABASE; do
+        >&2 echo "MySQL/MariaDB is unavailable - sleeping"
+        sleep 1
+    done
+}
+
+function update_ini() {
+    INI_FILE=$1
+    KEY=$2
+    NEW_VALUE=$3
+    sed -i "/^$KEY=/s/=.*/=$NEW_VALUE/" $INI_FILE
+}
+
+function update_ini_values() {
+    update_ini worldconfig.ini mysql_host $DATABASE_HOST
+    update_ini worldconfig.ini mysql_database $DATABASE
+    update_ini worldconfig.ini mysql_username $DATABASE_USER
+    update_ini worldconfig.ini mysql_password $USE_DATABASE_PASSWORD
+    update_ini worldconfig.ini chat_server_port $CHAT_SERVER_PORT
+    update_ini worldconfig.ini max_clients $MAX_CLIENTS
+}
+
+function symlink_client_files() {
+    ln -s /client/client/res/macros/ /app/res/macros
+    ln -s /client/client/res/BrickModels/ /app/res/BrickModels
+    ln -s /client/client/res/chatplus_en_us.txt /app/res/chatplus_en_us.txt
+    ln -s /client/client/res/names/ /app/res/names
+    ln -s /client/client/res/maps/ /app/res/maps
+    ln -s /client/client/locale/locale.xml /app/locale/locale.xml
+}
+
+function fdb_to_sqlite() {
+    echo "Run fdb_to_sqlite"
+    python3 /app/utils/fdb_to_sqlite.py /client/client/res/CDClient.fdb --sqlite_path /app/res/CDServer.sqlite 
+
+    (
+        cd /app/migrations/cdserver
+        readarray -d '' entries < <(printf '%s\0' *.sql | sort -zV)
+        for entry in "${entries[@]}"; do
+            echo "Execute $entry"
+            sqlite3 ../../res/CDServer.sqlite < $entry
+        done
+    )
+}
+
+function run_db_migrations() {
```

Comment:
No real need to do this. If you add the sql files under `/docker-entrypoint-initdb.d` MariaDB will automatically load them at container startup. Read about it here (https://hub.docker.com/_/mysql/) under the header "Initializing a fresh instance".

Reply:
Sure, but I am not certain I will use it. Because this would prevent the plan with an image we could push to docker hub.

Reply:
I mount it as a volume. That way no need to build it in to the final image.

Reply:
Created a PR for this change: https://github.com/TheNoim/DarkflameServer/pull/2